<launch>

  <!-- Set up visualisation -->
  <node pkg="tf" type="static_transform_publisher" name="camera_init_to_map"  args="0 0 0 0 0 0 1 map my_frame 10" />

  <arg name="rviz" default="true" />
  <node pkg="sl_sensor_visualise" type="pose_stamped_to_path_node" name="pose_stamped_to_path_node" output="screen">
    <param name="subscribe_topic" value="registered_pose" />
    <param name="publish_topic" value="path" />
  </node>

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find sl_sensor_visualise)/rviz/pairwise_registration.rviz" />
  </group>

  <!-- File information -->
  <param name="stitch_pc_directory" type="str" value="/home/ltf/rosbag_processing/input_pc/" /> <!-- Folder where the point clouds that will ultimately be registered and saved -->
  <param name="registration_pc_directory" type="str" value="/home/ltf/rosbag_processing/input_pc/" /> <!-- Folder where the point clouds will be used for transformation estimation -->
  <param name="output_pc_directory" type="str" value="/home/ltf/rosbag_processing/output_pc/" /> <!-- Folder where registered point clouds will be saved -->
  <param name="header_name" type="str" value="log_" />
  <param name="row_min" type="int" value="0" />  
  <param name="row_max" type="int" value="100000" /> 

  <!-- Setup ICP Registration-->
  <param name="pc_algo" type="str" value="lpm_icp" />
  <param name="leafsize" type="double" value="0.000" /> <!-- subsmpling method is specified in the yaml file -->
  <param name="yaml_directory" type="str" value="$(find sl_sensor_registration)/lpm_yaml/pairwise_registration_rosbag.yaml" />
  <node pkg="sl_sensor_registration" type="pairwise_registration_rosbag_node" name="pairwise_registration_rosbag_node" output="screen"/>
  

</launch>
